invenio_rdm_records/services/permissions.py

```diff
@@ -82,9 +83,15 @@ class RDMRecordPermissionPolicy(RecordPermissionPolicy):
         IfRestricted("record", then_=can_view, else_=can_all),
     ]
     # Allow reading the files of a record
-    can_read_files = [
+    can_read_files_owner = [RecordOwners(), SecretLinks("view"), SystemProcess()]
+    can_read_files_community = [
```

Comment:
I am a bit confused by reading the code.

`can_read_files_community` -> `IfRestricted("files", ...)`
`can_read_files` -> `IfRestrictedForCommunity`("files", ...)

The logic seems to be correct based on the issue's description though.
Maybe a small code comment could increase the readabilty.

Reply:
`can_read_files_community` keeps the same permission logic as it was before and is called in `else_` of `can_read_files` which will be valuated in case the community has access to the files of the record.
I understand can be confusing, will add the comment!